functions/sg/main.py

```diff
@@ -0,0 +1,107 @@
+import os
+import re
+import requests
+
+WFL_URL = os.environ.get('WFL_URL')
+CROMWELL_URL = os.environ.get('CROMWELL_URL')
+WORKLOAD_PROJECT = os.environ.get('WORKLOAD_PROJECT')
+OUTPUT_BUCKET = os.environ.get('OUTPUT_BUCKET')
+PATH_PATTERN = os.environ.get('PATH_PATTERN')
+
+assert WFL_URL is not None, 'WFL_URL is not set'
+assert CROMWELL_URL is not None, 'CROMWELL_URL is not set'
+assert WORKLOAD_PROJECT is not None, 'WORKLOAD_PROJECT is not set'
+assert OUTPUT_BUCKET is not None, 'OUTPUT_BUCKET is not set'
+
+
+def get_auth_headers():
+    scopes = ','.join([
+        'https://www.googleapis.com/auth/cloud-platform',
+        'https://www.googleapis.com/auth/userinfo.email',
+        'https://www.googleapis.com/auth/userinfo.profile'
+    ])
+    metadata_url = (
+        'http://metadata.google.internal/computeMetadata/v1/'
+        f'instance/service-accounts/default/token?scopes={scopes}'
+    )
+    metadata_headers = {'Metadata-Flavor': 'Google'}
+    r = requests.get(metadata_url, headers=metadata_headers)
+    r.raise_for_status()
+    access_token = r.json()['access_token']
+    headers = {
+        'Authorization': 'Bearer {}'.format(access_token)
+    }
+    return headers
+
+
+def filter_for_input_paths(pattern, event_file):
+    if pattern is None or re.search(pattern, event_file) is not None:
+        return event_file
+    else:
```

Comment:
You could get rid of the else block here, since the function will return None either way